ops/testing.py

```diff
@@ -311,17 +365,17 @@ def begin_with_initial_hooks(self) -> None:
                     app_name = self._backend._relation_app_and_units[rel_id]["app"]
                     self._emit_relation_created(relname, rel_id, app_name)
         if self._backend._is_leader:
-            self._charm.on.leader_elected.emit()
+            my_charm.on.leader_elected.emit()
         else:
-            self._charm.on.leader_settings_changed.emit()
-        self._charm.on.config_changed.emit()
-        self._charm.on.start.emit()
+            my_charm.on.leader_settings_changed.emit()
+        my_charm.on.config_changed.emit()
+        my_charm.on.start.emit()
```

Comment:
Agree that casting this to my_charm feels awkward. It attenuates my sense of where this value actually comes from.

(It would almost be easier if it were just called 'charm' -- it's a common pattern to do something like foo = self._foo. It feels weirder to do my_foo = self._foo.)